sqlglot/dialects/tsql.py

```diff
@@ -37,11 +106,52 @@ class Parser(Parser):
             "CHARINDEX": exp.StrPosition.from_arg_list,
         }
 
-        def _parse_convert(self):
+        def _parse_convert(self, strict):
             to = self._parse_types()
             self._match(TokenType.COMMA)
             this = self._parse_field()
-            return self.expression(exp.Cast, this=this, to=to)
+
+            var_length_datatype = [
```

Comment:
make this a constant outside of the function

Reply:
Placed it within the parser class now.